### PR TITLE
implement darkmode on the right-nav of the Learn/guide page

### PR DESCRIPTION
**Pull request by oyenuga17**
@Sayo @Puchangan please help review

### CODE REVIEW

File: src/ocamlorg_frontend/components/toc.eml [hunk 1]
```diff
@@ -29,8 +29,8 @@ let render (t : t) =
             <ol>
               <% items |> List.iter begin fun item -> %>
               <li>
-                <a href="<%s item.href %>" class="text-legacy-default py-1 md:py-0 pl-5 pr-1 block transition-colors border-l-4 border-t-2 border-b-2 border-r-2 border-transparent rounded"
-                  :class='isWithin(scrollY, "<%s item.href %>", sectionYPositions) ? "font-semibold border-l-primary text-primary bg-legacy-primary-200": "hover:text-primary"'
+                <a href="<%s item.href %>" class="text-content dark:text-dark-content py-1 md:py-0 pl-5 pr-1 block transition-colors border-l-4 border-t-2 border-b-2 border-r-2 border-transparent rounded"
+                  :class='isWithin(scrollY, "<%s item.href %>", sectionYPositions) ? "font-semibold border-l-primary dark:border-l-dark-primary text-primary dark:text-dark-primary bg-primary_nav_block_hover_10 dark:bg-primary_nav_block_hover_10": "hover:text-primary dark:hover:text-dark-primary"'
```
Comment (SaySayo) [inline src/ocamlorg_frontend/components/toc.eml]: I'm a bit unclear why the light and dark mode colors are both `bg-primary_nav_block_hover_10` is that according to the figma design?

File: src/ocamlorg_frontend/components/toc.eml [hunk 2]
```diff
@@ -15,12 +15,12 @@ let render (t : t) =
...
     <% (match t with [] ->  %>
     <% | _ -> %>
-    <div class="font-semibold text-gray-500 text-sm mb-4">On This Page</div>
+    <div class="font-bold text-title dark:text-dark-title text-sm mb-4">On This Page</div>
     <ol class="leading-6 text-sm border-l">
       <% t |> List.iter begin fun item -> %>
         <li>
-          <a href="<%s item.href %>" class="text-legacy-default py-1 pl-2 pr-1 block transition-colors border-l-4 border-t-2 border-b-2 border-r-2 border-transparent rounded"
-            :class='isWithin(scrollY, "<%s item.href %>", sectionYPositions) ? "font-semibold border-l-primary text-primary bg-legacy-primary-200": "hover:text-primary"'
+          <a href="<%s item.href %>" class="text-title dark:text-dark-title py-1 pl-2 pr-1 block transition-colors border-l-4 border-t-2 border-b-2 border-r-2 border-transparent rounded"
```
Comment (SaySayo) [inline src/ocamlorg_frontend/components/toc.eml]: Also why are there different text classes for the links, it is `text-title` here and `text-content` below for the same `text-legacy-default` class

Comment (oyenuga17) [reply src/ocamlorg_frontend/components/toc.eml]: Ugh! thankyou I have now fixed it

**Comment by oyenuga17** [comment]
closing this as changes have been implemented in #1913

**Pull request closed without merging**